### PR TITLE
Add short plugin overview and steps to add an API to contribution docs

### PR DESCRIPTION
**Pull request by jwilander**
Wanted to get us started on contribution docs for plugins.

### CODE REVIEW

File: site/content/contribute/server/plugins.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Check out the [`plugin` package](https://github.com/mattermost/mattermost-server/tree/master/plugin) and the [plugin_* files in the `app` package](https://github.com/mattermost/mattermost-server/tree/master/app) for the code, and [mattermost-plugin-demo](https://github.com/mattermost/mattermost-plugin-demo) for an example plugin.
+
+## Adding an API 
+
+To add a plugin API you need to add the signature of your new method to the [API interface](https://github.com/mattermost/mattermost-server/blob/master/plugin/api.go). You then need to implement the API in the [plugin_api.go](https://github.com/mattermost/mattermost-server/blob/master/app/plugin_api.go) of the `app` package. Finally, you would need to run `make pluginapi` to generate the RPC glue code needed for your new API.
```
Comment (jasonblais) [inline site/content/contribute/server/plugins.md]: Are any (unit) tests added?

Comment (jwilander) [reply site/content/contribute/server/plugins.md]: Testing of the wireframe is covered by the auto-generated code. For more complex API implementations we'll want separate tests but that'll be the special case so we'll ask for it case-by-case generally

File: site/content/contribute/server/plugins.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+In the future there will be another component of plugins for the mobile apps, likely a React Native bundle.
```
Comment (crspeller) [inline site/content/contribute/server/plugins.md]: Maybe add "for mobile". Some people interested might not know that our mobile app is React Native.

Comment (jwilander) [reply site/content/contribute/server/plugins.md]: It does say "for the mobile apps" right before, is that not good enough?

Comment (enahum) [reply site/content/contribute/server/plugins.md]: It is clear to me.

One thing that I’m thinking is that we should add somewhere that if you are making a plugin to change the format For a custom post type the mobile apps won’t render them as expected but instead will render the raw message formatted with our standard formatter?

**Pull request merged**